src/backend/main.ts

```diff
@@ -706,10 +717,31 @@ ipcMain.once('loadingScreenReady', () => {
 
 ipcMain.once('frontendReady', async () => {
   logInfo('Frontend Ready', LogPrefix.Backend)
+  const currentVersion = app.getVersion()
+  const lastVersion = configStore.get('appVersion', '')
+  const walletStateIsConnected = configStore.get(
+    'walletState.isConnected',
+    false
+  )
+
   await initExtension(hpApi)
-  // wait for mm SW to initialize
-  await wait(5000)
-  ipcMain.emit('reloadApp')
+
+  // Only reload the app if a wallet is not connected
+  // or the app version has changed
+  if (
+    !walletStateIsConnected ||
```

Comment:
which edge case does this address?

Reply:
Joe found out that it is not possible to import browser profiles if we do not reload. But looking at the logic not, this issue was maybe caused by the order on which I was adding the initExtension before. I will ask him to test again and will remove this check for now.

Reply:
OK it seems we did not need this.